File: app/src/main/java/ru/anasoft/calculator/SettingsActivity.java
package ru.anasoft.calculator;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.RadioGroup;

public class SettingsActivity extends AppCompatActivity {

    int numberTheme;
    RadioGroup radioGroup;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        numberTheme = getIntent().getExtras().getInt(Constants.MY_THEME);
        radioGroup = findViewById(R.id.radioGroupTheme);

        initRadioGroup();
        initButtonOK();
    }

    private void initRadioGroup() {
        switch (numberTheme) {
            case 1:
                radioGroup.check(R.id.radioButtonTheme1);
                break;
            case 2:
                radioGroup.check(R.id.radioButtonTheme2);
                break;
            case 3:
                radioGroup.check(R.id.radioButtonTheme3);
                break;
        }

    }

    private void initButtonOK() {
        Button buttonOK = findViewById(R.id.buttonOK);
        buttonOK.setOnClickListener(v -> {
            Intent intentResult = new Intent();
            intentResult.putExtra(Constants.MY_THEME, getMyChoice());
            setResult(RESULT_OK, intentResult);
            finish();
        });
    }

    private int getMyChoice() {
         switch (radioGroup.getCheckedRadioButtonId()) {
            case (R.id.radioButtonTheme1):
                return 1;
            case (R.id.radioButtonTheme2):
                return 2;
            case (R.id.radioButtonTheme3):
                return 3;
        }
        return 1;
    }

}

File: app/src/main/java/ru/anasoft/calculator/SettingsActivity.kt
package ru.anasoft.calculator

import androidx.appcompat.app.AppCompatActivity
import android.widget.RadioGroup
import android.os.Bundle
import android.content.Intent
import android.widget.Button

class SettingsActivity : AppCompatActivity() {
    private var numberTheme = 0
    private var radioGroup: RadioGroup? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        numberTheme = intent.extras!!.getInt(Constants.MY_THEME)
        radioGroup = findViewById(R.id.radioGroupTheme)
        initRadioGroup()
        initButtonOK()
    }

    private fun initRadioGroup() {
        when (numberTheme) {
            1 -> radioGroup!!.check(R.id.radioButtonTheme1)
            2 -> radioGroup!!.check(R.id.radioButtonTheme2)
            3 -> radioGroup!!.check(R.id.radioButtonTheme3)
        }
    }

    private fun initButtonOK() {
        val buttonOK = findViewById<Button>(R.id.buttonOK)
        buttonOK.setOnClickListener {
            val intentResult = Intent()
            intentResult.putExtra(Constants.MY_THEME, myChoice)
            setResult(RESULT_OK, intentResult)
            finish()
        }
    }

    private val myChoice: Int
        get() {
            when (radioGroup!!.checkedRadioButtonId) {
                R.id.radioButtonTheme1 -> return 1
                R.id.radioButtonTheme2 -> return 2
                R.id.radioButtonTheme3 -> return 3
            }
            return 1
        }
}